tracing.go

```diff
@@ -6,176 +6,273 @@
 package main
 
 import (
+	"bytes"
 	"context"
-	"io"
-
-	opentracing "github.com/opentracing/opentracing-go"
-	"github.com/uber/jaeger-client-go/config"
-)
-
-const (
-	jaegerAgentHost = "127.0.0.1"
-
-	// This is the default.
-	jaegerAgentPort = "6831"
+	"fmt"
+	"encoding/binary"
+	"encoding/gob"
+	"net"
+
+	gen "github.com/mcastelino/custom_exporter/jaeger"
+	"github.com/mdlayher/vsock"
+	"go.opencensus.io/trace"
 )
 
-// agentSpan implements opentracing.Span
+// agentSpan defines span by using opencensus
 type agentSpan struct {
-	span opentracing.Span
+	span *trace.Span
 }
 
 // The first trace span
 var rootSpan *agentSpan
 
-// Implements jaeger-client-go.Logger interface
-type traceLogger struct {
+// startSpan starts a new child span of the current span in the context
+func startSpan(ctx context.Context, name string) (context.Context, *agentSpan) \
+{
+	var a agentSpan
+	ctx, a.span = trace.StartSpan(ctx, name)
+	return ctx, &a
 }
 
-func (a *agentSpan) setTag(key string, value interface{}) *agentSpan {
-	a.span.SetTag(key, value)
-	return a
+// spanFromContext returns the span stored in a context
+func spanFromContext(ctx context.Context) *agentSpan {
+	var a agentSpan
+	a.span = trace.FromContext(ctx)
+	return &a
 }
 
-func (a *agentSpan) finish() {
-	a.span.Finish()
+// spanStartSpanWithRemoveParent starts a new child span of the span from the given parent
+func spanStartSpanWithRemoteParent(ctx context.Context, name string, parent tra\
+ce.SpanContext) (context.Context, *agentSpan) {
+	a := &agentSpan{}
+	ctx, a.span = trace.StartSpanWithRemoteParent(ctx, name, parent)
+	return ctx, a
 }
 
-func (a *agentSpan) tracer() agentTracer {
-	return agentTracer{tracer: a.span.Tracer()}
+// spanNewContext returns a new context with the givien span attached
+func spanNewContext(parent context.Context, s *agentSpan) context.Context {
+	return trace.NewContext(parent, s.span)
 }
 
-type agentTracer struct {
-	tracer opentracing.Tracer
+// setTag sets the name of the span, if it is recording events
+func (a *agentSpan) setTag(name string, value interface{}) {
+	a.span.SetName(fmt.Sprintf("%s=%v" ,name, value))
```

Comment:
This doesn't look correct to me. According to the API, `SetName()` sets the name of the trace itself. But tags in openTracing are "name=value" pairs. Each trace can have zero or more tags. I think we should be using https://godoc.org/go.opencensus.io/tag?

Reply:
@jodh-intel , looking the API for tag on opencensus, I can't find a function where we have as input(key, value) like we have in opentracing, if you see this
```
func MustNewKey(name string) Key
func NewKey(name string) (Key, error)
func (k Key) Name() string
```
they want as an input a `name` and they return a `key` and `error` which looks very similar to the current implementation that I have, wdyt?

Reply:
@jodh-intel , I change the tag implementation, thanks